File: spengerclient/src/app/models/Medication.ts
export class Medication {
    public static statusCode: Array<string> = ["active", "inactive", "entered-in-error"]
    constructor(
        public id: string = "",
        public resourceType: string = "Medication",
        public identifier?: Array<Identifier>,
        public code: CodeableConcept = new CodeableConcept([], ""),
        public status: string = Medication.statusCode[0],
        public manufacturer?: Reference,
        public form?: CodeableConcept,
        public amount?: Ratio,
        public ingredient?: Array<Ingredient>,
        public batch: Batch = new Batch()
    ) { }
}
export class Identifier {
    public static useCode: Array<string> = ["usual", "official", "temp", "secondary", "old"]
    constructor(
        public use: string = "",
        public value: string = ""
    ) { }
}
export class CodeableConcept {
    constructor(
        public coding: Array<Coding>,
        public text: string = ""
    ) { }
}
export class Coding {
    constructor(
        public system: string,
        public code: string,
        public display: string
    ) { }
}
export class Reference {
    constructor(
        public reference: string,
        public display: string
    ) { }
}
export class Ratio {
    constructor(
        public numerator: Quantity,
        public denominator: Quantity
    ) { }
}
export class Quantity {
    public static useCode: Array<string> = ["<", "<=", ">=", ">"]
    constructor(
        public value: number,
        public comparator: string,
        public unit: string,
        public system: string,
        public code: string
    ) { }
}
export class Ingredient {
    constructor(
        public itemCodeableConcept: CodeableConcept,
        public itemReference: Reference = new Reference("", ""),
        public isActive: boolean,
        public strengh: Ratio
    ) { }
}
export class Batch {
    constructor(
        public lotNumber: string = "",
        public expirationDate: Date = new Date(2025, 1, 1),
    ) { }
}

File: src/main/java/at/spengergasse/spengeronfhir/entities/Medication.java
package at.spengergasse.spengeronfhir.entities;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "m_medication")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@SuperBuilder
public class Medication extends DomainResource{

    @OneToMany(cascade = CascadeType.ALL,fetch = FetchType.EAGER)
    @JoinColumn(name = "i_m_id", referencedColumnName = "id")
    private List<Identifier> identifier = new ArrayList<Identifier>();

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "m_c_id")
    private CodeableConcept code;

    public enum StatusCode{
        active,
        inactive,
        @JsonProperty("entered-in-error") entered_in_error
    }

    @Enumerated(EnumType.STRING)
    @Column(name = "m_status")
    private StatusCode status;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "m_m_id")
    private Reference manufacturer;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "m_f_id")
    private CodeableConcept form;

    @OneToOne(cascade = CascadeType.ALL,fetch = FetchType.EAGER)
    @JoinColumn(name = "m_a_id")
    private Ratio amount;

    @OneToMany(cascade = CascadeType.ALL,fetch = FetchType.EAGER)
    @JoinColumn(name = "in_m_id", referencedColumnName = "id")
    private List<Ingredient> ingredient = new ArrayList<Ingredient>();

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "m_b_id")
    private Batch batch;
}
